Card/src/main/java/com/paypal/android/card/CardClient.kt

```diff
@@ -45,102 +46,61 @@ class CardClient internal constructor(
         configuration: CoreConfig,
     ) : this(activity, CardAPI(API(configuration)))
 
+    init {
+        activity.lifecycle.addObserver(lifeCycleObserver)
+    }
+
     /**
      * Confirm [Card] payment source for an order.
      *
      * @param orderId [String] order id to confirm
      * @param cardRequest [CardRequest] for requesting an order approval
-     * @param threeDSecureRequest [threeDSecureRequest] to run transaction with 3DS
-     * @param coroutineContext [CoroutineContext] to specify in which context to run api calls
      */
     @JvmOverloads
     fun approveOrder(
+        activity: FragmentActivity,
         orderId: String,
         cardRequest: CardRequest,
-        threeDSecureRequest: ThreeDSecureRequest? = null,
-        coroutineContext: CoroutineContext? = Dispatchers.Default
+        context: CoroutineContext = Dispatchers.IO
     ) {
-        coroutineContext?.also { context ->
-            lifeCycleObserver = CardLifeCycleObserver(this)
-            lifeCycleObserver?.let { activity.lifecycle.addObserver(it) }
-            currentCoroutineContext = context
-            CoroutineScope(context).launch {
-                try {
-                    confirmPaymentSource(orderId, cardRequest, threeDSecureRequest)
-                } catch (e: PayPalSDKError) {
-                    approveOrderListener?.failure(e)
-                    clearClient()
-                }
+        CoroutineScope(context).launch {
+            try {
+                confirmPaymentSource(activity, orderId, cardRequest)
+            } catch (e: PayPalSDKError) {
+                approveOrderListener?.onApproveOrderFailure(e)
+                clearClient()
             }
         }
     }
 
-    /**
-     * Creates order and confirms [Card] payment source for said order. Its private until phase 2
-     *
-     * @param orderRequest [String] order to create
-     * @param cardRequest [CardRequest] for requesting an order approval
-     * @param threeDSecureRequest [threeDSecureRequest] to run transaction with 3DS
-     * @param coroutineContext [CoroutineContext] to specify in which context to run api calls
-     */
-    private fun createAndApproveOrder(
```

Comment:
This is definetly going to be in phase 2. If our "policy" is to keep the main branch as clean as possible, im going to put a PR for this just to have it ready when we need it.

Reply:
Yeah that's the thought here. We want to decrease the scope of the PR to keep it as focused as possible.